database/triggers.py

```diff
@@ -214,19 +214,31 @@ def generate_update_lifecycle_status_triggers():
             )
             trgs.append(trg)
 
-    # Finally, we want to update general regulations as well:
+    # Finally, we want to update regulations belonging to a general regulation
+    # groups as well:
     for regulation_table in plan_regulation_tables:
         trgfunc_signature = f"trgfunc_plan_{regulation_table}_update_lifecycle_status()"
         trgfunc_definition = f"""
-        RETURNS TRIGGER AS $$
-        BEGIN
-            UPDATE hame.{regulation_table}
-            SET lifecycle_status_id = NEW.lifecycle_status_id
-            WHERE (plan_regulation_group_id = NEW.plan_regulation_group_id
-            AND lifecycle_status_id = OLD.lifecycle_status_id);
-            RETURN NEW;
-        END;
-        $$ language 'plpgsql'
+            RETURNS TRIGGER AS $$
+            BEGIN
+                UPDATE hame.{regulation_table}
+                SET lifecycle_status_id = NEW.lifecycle_status_id
+                WHERE
+                    lifecycle_status_id = OLD.lifecycle_status_id
+                    AND EXISTS ( -- Only update in case of general regulation group
```

Comment:
Should we actually update them all instead, if we have plan_id in all regulation groups?

Then we could remove the separate trigger from plan object to plan regulation completely.

This would mean all statuses are updated using plan_id, and changes in plan object lifecycle statuses would no longer propagate at all.

I believe currently, if plan object lifecycle status is changed, the regulation lifecycle status is always changed even if the regulation is in multiple plan objects? Perhaps we should not do that?

Reply:
This will be resoveld in #403. This PR is only for general regulation groups so didn't want to include those changes here.